### ISSUE TITLE
S3 methods on R < 3.6.0

### ISSUE THREAD

**Issue by lionel-** [r-quantities/errors#37]
{errors} lazily exports S3 methods with:

```r
#' @rawNamespace if(getRversion() >= "3.6.0") {
#'   S3method(pillar::type_sum, errors)
#'   S3method(pillar::pillar_shaft, errors)
#' } else {
#'   export(type_sum.errors)
#'   export(pillar_shaft.errors)
#' }
```

This isn't a robust way because on R < 3.6 the methods will be found only when the package is attached to the search path with `library(errors)`, not when the package is merely loaded. I recommend copying `vctrs::s3_register()` in your namespace and registering in the `.onLoad()` hook, if you'd like to support old R versions. This way it will work even in the case users create errors objects without the package attached, e.g. with the namespace qualifier `errors::set_errors()`.

**Comment by lionel-** [comment]
Also the vctrs methods for double dispatch generics cannot be exported in this way, because we only look in the global environment, not the search path (consistently with recent versions of R).

**Comment by Enchufa2** [comment]
I don't particularly care for old versions of R. The plan was to get rid of that if/else as soon as R 4 matures. But ok, I suppose that supporting R 3.5.x for some time is reasonable.

**Comment by Enchufa2** [comment]
Closed by #38.
